gobblin-service/src/main/java/org/apache/gobblin/service/modules/scheduler/GobblinServiceJobScheduler.java

```diff
@@ -307,6 +282,14 @@ public void onUpdateSpec(Spec updatedSpec) {
     }
   }
 
+  public void becomingActive() {
```

Comment:
why are state transitions handled inside GobblinServiceJobScheduler? Are these states of scheduler or that of GobblinServiceManager?

Reply:
this is a callback. state belongs to GobblinServiceManager and the GobblinServiceJobScheduler is just a listener of the state transition and do some actions upon the transition